### DISCUSSION TITLE
在使用库的时候遇到了一些问题

### DISCUSSION THREAD

**Discussion by Channing-Huang** [jd-opensource/drip-form#181] [Q&A] [answered]
1、在npm安装好了 drip-from后，还需要手动自己安装ajv和antd
2、在按照[文档](https://jdfed.github.io/drip-form/docs/use/uiComponents)引入 样式'@jdfed/drip-form-theme-antd/dist/index.css'， 并没有按照antd的组件模式渲染， 手动引入antd的css, 样式还是存在问题。

**Comment by mengshang918** [comment]
1. antd需要手动安装，ajv应该不需要
   form相关主题包都是external的相应主题的
   ajv不需要手动安装，如果你那边需要手动安装，可以提供一个仓库，我看下。
2. 样式问题，可以看下是否引入了drip-form自带的样式

    ```
     import '@jdfed/drip-form/dist/index.css'
     import '@jdfed/drip-form-theme-antd/dist/index.css'
   ```

**Comment by Channing-Huang** [comment, marked as answer]
1. ajv我手动卸载后再次安装drip-from正常了，没能复现。
2. 样式问题引入上述两个样式文件，再额外引入antd.css还是会出现异常，同时根据文档引入的自定义组件也不能展示，dom是空的。
异常代码：[代码链接](https://codesandbox.io/s/frosty-wood-1rqpo3?file=/src/App.js) ，（我这边codesandbox一直在loading, 不知道最终渲染是不是和我本地下方图片渲染的一致）
异常样式如下：
![image](https://user-images.githubusercontent.com/27494499/161233359-ab978dc0-d388-4a66-881c-8c3e654568a6.png)

**Reply by mengshang918** [reply]
我这边写了一个例子，可以参考下[codeSandbox](https://codesandbox.io/s/vite-antd-forked-5k27f?file=/packages/project/src/demo.schema.ts)
1. codeSandbox一直loading
@jdfed/drip-form-theme-antd打包的代码引入的antd是less形式的，所以需要配置less-loader。
这边v0.8.0的规划里有这个问题的解决方案了。后续会默认多种打包形式，可以参考[build 主题包多种打包形式](https://github.com/orgs/JDFED/projects/2/views/7)
2. 样式和自定义组件不展示问题
    很抱歉，我们的文档示例有问题，自定义组件的customComponents的key值需要和unitedSchema中的fieldKey相同。并且自定义组件的type都是custom （具体可以参考我上面发的codesandbox示例）
3. 样式问题
    select组件，在antd主题内部，是没有默认设置宽度的（为了保持和option中的选项保持一致）。
   我们这边可以在ui中加width：100，也可以根据需要添加其他宽度值（比如：%、rem、vw等）

文档部分示例目前缺少一定测试，后续文档中会有每个例子的codeSandbox示例。在v0.8.0发布之后，q2会对整个文档进行重构。
对文档有建议也可以讨论，感谢～
我们也欢迎随时提pr修改文档错误内容～

**Reply by Channing-Huang** [reply]
我这边去试下，万分感谢！
